front/commondropdown.php

```diff
@@ -2,10 +2,9 @@
 
 include "../../../inc/includes.php";
 $path = PLUGINFIELDS_FRONT_PATH . '/' . $_REQUEST['ddtype'] . '.php';
```

Comment:
This test was made tio ensure that this script is not used to execute code that is located outside the plugin directory.

```suggestion
if (preg_match('/[a-z]/i', $_REQUEST['ddtype']) !== 1) {
   throw new \RuntimeException(sprintf('Invalid itemtype "%"', $_REQUEST['ddtype']));
}
$path = PLUGINFIELDS_FRONT_PATH . '/' . $_REQUEST['ddtype'] . '.php';
```

Reply:
Seems like a good way to do it :+1:
If understand correctly you check that the file name contains only text so that you can't go to another folder with `../../..`.